### DISCUSSION TITLE
state var dict, no longer a dict?

### DISCUSSION THREAD

**Discussion by gsal** [reflex-dev/reflex#560] [General]
Hi, 

I have a state var  ` mydict: dict = {"one":1 , "two":2 }`  and I cannot do the typical `mydict.keys()`  on it....as if it is not actually a dict, after all....this is rather disorienting. 

I mean, it's nice Pynecone is making web-development easier, but, shouldn't it do it by extending Python without taking away Python? 

In the mean time, it seems that we would need a lot more documentation, I keep running into aspects of pynecone that are not being published. 

So, how do I get the keys of my dictionary?

**Comment by Lendemor** [comment]
If you do it within an event handler or computed vars you should be able to access `mydict.keys()`.

If the dict is not meant to be rendered you can also use backend variables.

Can't tell what would be the best option for you without seeing the code you tried.

**Comment by gsal** [comment]
OK, thanks for those two hints. 
Indeed, I was trying to use dict.keys() as the first argument to pc.foreach()
